catalog/src/main/kotlin/com/adevinta/spark/catalog/examples/samples/tokens/colors/ColorSample.kt

```diff
@@ -41,71 +43,88 @@ import androidx.compose.ui.tooling.preview.Preview
-            .verticalScroll(scrollState)
             .fillMaxWidth(),
-        maxItemsInEachRow = 3,
     ) {
-        ColorItem(SparkTheme.colors.main, "main")
-        ColorItem(SparkTheme.colors.mainContainer, "main Container")
-        ColorItem(SparkTheme.colors.mainVariant, "main Variant")
-        ColorItem(SparkTheme.colors.support, "support")
-        ColorItem(SparkTheme.colors.supportContainer, "support Container")
-        ColorItem(SparkTheme.colors.supportVariant, "support Variant")
-        ColorItem(SparkTheme.colors.accent, "accent")
-        ColorItem(SparkTheme.colors.accentContainer, "accent Container")
-        ColorItem(SparkTheme.colors.accentVariant, "accent Variant")
-        ColorItem(SparkTheme.colors.basic, "basic")
-        ColorItem(SparkTheme.colors.basicContainer, "basic Container")
-        ColorItem(SparkTheme.colors.background, "background")
-        ColorItem(SparkTheme.colors.backgroundVariant, "backgroundVariant")
-        ColorItem(SparkTheme.colors.surface, "surface")
-        ColorItem(SparkTheme.colors.surfaceInverse, "surface inverse")
-        ColorItem(SparkTheme.colors.outline, "outline")
-        ColorItem(SparkTheme.colors.outlineHigh, "outline High")
-        ColorItem(SparkTheme.colors.success, "success")
-        ColorItem(SparkTheme.colors.successContainer, "success Container")
-        ColorItem(SparkTheme.colors.alert, "alert")
-        ColorItem(SparkTheme.colors.alertContainer, "alert Container")
-        ColorItem(SparkTheme.colors.error, "error")
-        ColorItem(SparkTheme.colors.errorContainer, "error Container")
-        ColorItem(SparkTheme.colors.info, "info")
-        ColorItem(SparkTheme.colors.infoContainer, "info Container")
-        ColorItem(SparkTheme.colors.neutral, "neutral")
-        ColorItem(SparkTheme.colors.neutralContainer, "neutral Container")
+        items(tokens) { token ->
+            Row {
+                token.forEach {
+                    ColorItem(it)
+                }
+            }
+        }
     }
 }
 
 @Composable
-private fun RowScope.ColorItem(color: Color, colorName: String) {
+private fun RowScope.ColorItem(color: KProperty0<Color>) {
     Surface(
         modifier = Modifier
             .padding(8.dp)
-            .sizeIn(minWidth = 104.dp, minHeight = 104.dp)
+            .heightIn(104.dp)
             .weight(1f),
+        color = color.get(),
         shape = SparkTheme.shapes.extraLarge,
-        color = color,
         border = BorderStroke(2.dp, SparkTheme.colors.onBackground),
     ) {
         Box(
             modifier = Modifier.padding(8.dp),
             contentAlignment = Alignment.Center,
         ) {
             Text(
-                text = colorName,
+                text = color.name,
                 style = SparkTheme.typography.body2,
                 textAlign = TextAlign.Center,
             )
         }
     }
 }
 
+private val tokensColors
+    @Composable
+    get() = with(SparkTheme.colors) {
+        listOf(
```

Comment:
nit: we probably can directly ask for the complete list of properties instead of keeping this manual listing.

Reply:
Yeah I've been wondering about that but we would have to show the deprecated one as well until the end of Q3 :(

Reply:
Or `.filter()` the deprecated properties :)

Reply:
```kotlin
field.hasAnnotation<Deprecated>()
```